vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/Map.java

```diff
@@ -321,4 +327,44 @@ public double getZoom() {
     public void setZoom(double zoom) {
         getView().setZoom(zoom);
     }
+
+    /**
+     * Zooms and pans the map to fit the given features in the viewport. Uses a
+     * default padding of 50 pixels and animation duration of 400 milliseconds.
+     *
+     * @param features
+     *            the features to fit in the viewport
+     */
+    public void zoomToFit(List<Feature> features) {
+        zoomToFit(features, 50, 400);
+    }
+
+    /**
+     * Zooms and pans the map to fit the given features in the viewport.
+     *
+     * @param features
+     *            the features to fit in the viewport
+     * @param padding
+     *            padding in pixels to add around the features
+     * @param duration
+     *            animation duration in milliseconds, 0 for no animation
+     */
+    public void zoomToFit(List<Feature> features, int padding, int duration) {
+        ArrayNode featureIds = JacksonUtils.createArrayNode();
+        features.forEach(feature -> featureIds.add(feature.getId()));
+
+        ObjectNode options = JacksonUtils.createObjectNode();
+        options.put("padding", padding);
+        options.put("duration", duration);
+
+        // In order to allow calling this method before the map, or the view
+        // that contains it, is attached to the UI, this needs to be delayed
+        // with beforeClientResponse to ensure that the client-side connector
+        // has been initialized beforehand.
+        getElement().getNode()
+                .runWhenAttached(ui -> ui.beforeClientResponse(this,
+                        context -> getElement().executeJs(
+                                "this.$connector.zoomToFit($0, $1)", featureIds,
+                                options)));
```

Comment:
Could not find a better approach. In some form this needs to run after `onAttach`, which is responsible for creating the connector.